src/vulkan-renderer/mesh_buffer.cpp

```diff
@@ -0,0 +1,84 @@
+#include "inexor/vulkan-renderer/mesh_buffer.hpp"
+
+namespace inexor::vulkan_renderer {
+MeshBuffer::MeshBuffer(MeshBuffer &&other) noexcept
+    : name(std::move(other.name)), vertex_buffer(std::move(other.vertex_buffer)), index_buffer(std::move(other.index_buffer)),
+      number_of_vertices(other.number_of_vertices), number_of_indices(other.number_of_indices) {}
+
+MeshBuffer::MeshBuffer(const VkDevice device, VkQueue data_transfer_queue, const std::uint32_t data_transfer_queue_family_index,
+                       const VmaAllocator vma_allocator, std::string name, const VkDeviceSize size_of_vertex_structure, const std::size_t number_of_vertices,
+                       void *vertices, const VkDeviceSize size_of_index_structure, const std::size_t number_of_indices, void *indices)
+
+    // It's no problem to create the vertex buffer and index buffer before the corresponding staging buffers are created!.
+    : vertex_buffer(device, vma_allocator, name, size_of_vertex_structure * number_of_vertices,
+                    VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_VERTEX_BUFFER_BIT, VMA_MEMORY_USAGE_CPU_ONLY),
+      index_buffer(GPUMemoryBuffer(device, vma_allocator, name, size_of_index_structure * number_of_vertices,
+                                   VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_VERTEX_BUFFER_BIT, VMA_MEMORY_USAGE_CPU_ONLY)) {
+    assert(device);
+    assert(vma_allocator);
+    assert(!name.empty());
+    assert(size_of_vertex_structure > 0);
+
+    std::size_t vertex_buffer_size = size_of_vertex_structure * number_of_vertices;
```

Comment:
Consider using only one kind of variable name `foo_size` vs `size_of_foo`.

Reply:
The vulkan spec usually uses `vertex_count` instead of `number_of_vertices`.

Reply:
I will be using `size_of_vertex_buffer` this time. I will use more consistent naming in the upcoming pull requests.